File: Snippets/Sum_of_Digits/sod.js
//
// Return the sum of the digits of the set of given numbers.
//
function digit_sum (numbers) {
    let sum = 0
    numbers . forEach (number => {
        while (number > 0) {
            sum += number % 10
            number = Math . floor (number / 10)   
        }
    })
    return sum
}

function digit_sum1 (number) {
    let sum  =  0
    let base = 10
    while (number > 0) {
        sum   += number % base
        number = Math . floor (number / base)
    }
    return sum
}

File: Snippets/Sum_of_Digits/sod.java
public class main {
    private static int digit_sum (int n) {
        int sum  =  0;
        int base = 10;
        while (n > 0) {
            sum += n % base;
            n    = (int) Math . floor (n / base);
        }
        return (sum);
    }
}
